eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/parallel/set/ParallelCollectIterable.java

```diff
@@ -98,20 +98,20 @@ public V detect(Predicate<? super V> predicate)
     public <V1> UnsortedBagMultimap<V1, V> groupBy(Function<? super V, ? extends V1> function)
     {
         // TODO: Implement in parallel
-        return this.delegate.toBag().collect(this.function).groupBy(function);
+        return (UnsortedBagMultimap<V1, V>) this.delegate.toBag().collect(this.function).groupBy(function);
```

Comment:
You tried extracting all three intermediate results as local variables? Are you confident this isn't just a bug in javac?